harness/java/io/temporal/sdkfeatures/Main.java

```diff
@@ -60,6 +92,27 @@ public void run() {
       throw new RuntimeException("Client cert path must be specified since key path is");
     }
 
+    // Summary server
+    BufferedWriter writer = null;
+    try {
+      URI uri = new URI(summaryUri);
+      switch (uri.getScheme()) {
+        case "tcp":
+          Socket socket = new Socket(uri.getHost(), uri.getPort());
+          writer = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream(), "UTF-8"));
+          break;
+        case "file":
+          FileWriter fileWriter = new FileWriter(uri.getPath(), true);
+          writer = new BufferedWriter(fileWriter);
+          break;
```

Comment:
You should be closing things you open. Consider a try with resources, or better, we should have used a higher level protocol like HTTP.